src/components/TokenBalance.tsx

```diff
@@ -79,56 +74,65 @@ const TokenBalance: FC<TokenBalanceProps & TextProps> = ({
   tokenAmount,
   usdBalance,
   tokenSymbol,
-  withUSDBalance = false,
-  withSymbol = false,
   tokenDecimals,
   isLarge,
   tokenFormat,
   precision,
   higherPrecision,
-  withHigherPrecision = false,
+  children,
   ...restProps
 }) => {
   const { active } = useWeb3React()
   const shouldRenderTokenAmount = active
 
   const _tokenAmount = useMemo(() => {
-    return formatTokenAmount(tokenAmount || 0, tokenFormat, tokenDecimals)
+    return formatTokenAmount(
+      tokenAmount || 0,
+      tokenFormat,
+      tokenDecimals,
+      precision
+    )
   }, [tokenAmount, tokenFormat, tokenDecimals])
```

Comment:
Let's add `precision` to `useMemo` dependencies:

```suggestion
  }, [tokenAmount, tokenFormat, tokenDecimals, dependencies])
```

Reply:
Ref commit: 795d533a